internal/target/repository_host_set_test.go

```diff
@@ -0,0 +1,493 @@
+package target
+
+import (
+	"context"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/host/static"
+	"github.com/hashicorp/boundary/internal/iam"
+	"github.com/hashicorp/boundary/internal/kms"
+	"github.com/hashicorp/boundary/internal/oplog"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/proto"
+)
+
+func TestRepository_AddTargetHostSets(t *testing.T) {
```

Comment:
I realize this was just a refactoring (move to it's own file) but consider adding the corresponding tests I called out in the repository_credential_library_test.go file regarding adding host sets that are already attached and removing sets which aren't associated yet at all.  This will help ensure all SAD operation behaviors are consistent with each other.